doc/networking.md

```diff
@@ -0,0 +1,132 @@
+2. Ensure the orchestrator is assigned to the job.
+3. Ensure the current block falls within the valid range of `CreationBlock` ... `EndBlock` for the job.
+4. Verify that the signature comes from the broadcaster.
+
+The `TranscoderInfo` response contains:
+
+```protobuf
+// The orchestrator sends this in response to `GetTranscoder`, containing the
+// transcoder URI, associated credentials authorizing the broadcaster to
+// use the transcoder, and miscellaneous data related to the job.
+message TranscoderInfo {
+
+  // URI of the transcoder to use for submitting segments
+  string transcoder = 1;
+
+  // Signals the authentication method to expect within `credentials`. This
+  // field is opaque to the broadcaster, and should be passed to the transcoder.
+  string authType = 2;
+
+  // Credentials to verify the request has been authorized by an orchestrator.
+  // This field is opaque to the broadcaster.
+  string credentials = 3;
+
+  // Transcoded streamId list to update the master manifest on the broadcaster.
+  map<string, string> streamIds = 50;
+}
+```
+
+### Notes
+The authorization token will become more useful when the transcoder and orchestrator are separate entities. Additionally, the auth token can incorporate additional fields to avoid an on-chain lookup, such as the StreamID, token start time, and approximate job end time. This is stored as a base64-encoded, serialized protbuf struct.
+
+```protobuf
+// AuthToken is sent by the orchestrator and encoded in the `credentials` field
+// This record is opaque to the broadcaster and is only relevant between the
+// orchestrator and the transcoder.
+message AuthToken {
+
+  // Signature of the orchestrator over the remaining fields
+  bytes sig = 1;
+
+  int64 jobId = 10;
+}
+```
+
+## Broadcaster to Transcoder
+
+### POST `/segment`
+
+Invoked each by the broadcaster for each segment that needs to be transcoded. The transcoder address is taken from the `transcoder` field in `TranscoderInfo`.
+
+#### Required Headers:
+
+* **Authorization**
+Taken from the `authType` field in TranscoderInfo.
+* **Credentials**
+Taken from the `credentials` field in TranscoderInfo.
+* **Livepeer-Segment**
+Proves that the broadcaster generated this segment. Serialized protobuf struct, base64 encoded.
+```protobuf
+message SegData {
+  int64 seq  = 1;
+  bytes hash = 2;
+  bytes sig  = 3;
+}
+```
+* Content-Type
+`media/livepeer-videoblob`
+
+The composition of the body (and certain headers) varies based on the content-type. For the content-type of `media/livepeer-videoblob` , the body is composed of the bytes of the segment.
+
+Processing a `/segment` request consists of the following steps:
+
+1. Verify the credentials generated by the orchestrator
+2. Verify that the job hasn't expired.
```

Comment:
Just thinking about the scenario where the orchestrator and transcoder are physically distinct entities - this step would require the transcoder to have access to the ETH network if it is attempting to be self-sufficient by verifying the job expiration itself via a contract lookup and by having knowledge of the current block number. But, if it does not need to be self-sufficient (the trusted case), the auth token it receives from the orchestrator could contain the approximate job end time as you mentioned so the transcoder wouldn't need any access to the ETH network. It might not be very exact given varying block times, but perhaps it could be decent estimate...

Reply:
I think the idea of transcoders not needing to access the chain - and instead they just take the orchestrators directives - as appealing. But maybe I'm looking too far ahead to transcoder races, when right now we should just keep focus on this first implementation to get to a stable network.

Reply:
Yeah, I punted on the "off-chain transcoder" issue for now, because there are still some unresolved questions, such as deposit checking when a broadcaster has multiple concurrent jobs.